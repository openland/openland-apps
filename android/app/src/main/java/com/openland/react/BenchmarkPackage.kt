package com.openland.react

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class BenchmarkModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "BenchmarkModule"
    }

    @ReactMethod
    fun setConfig() {
    }
}