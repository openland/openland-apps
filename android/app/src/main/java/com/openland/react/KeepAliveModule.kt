package com.openland.react

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.jstasks.HeadlessJsTaskContext
//import com.facebook.react.modules.core.Timing
import java.util.concurrent.atomic.AtomicBoolean

class KeepAliveModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName(): String {
        return "RNKeepAlive"
    }

    override fun initialize() {
        
        // Hack Timers
        runOnUIThread {
            val headlessContext = HeadlessJsTaskContext.getInstance(reactApplicationContext)
            try {
                val mActiveTasks = HeadlessJsTaskContext::class.java.getDeclaredField("mActiveTasks")
                mActiveTasks.isAccessible = true
                val tasks = mActiveTasks.get(headlessContext) as MutableSet<Int>
                tasks.add(-1)
            } catch (e: NoSuchFieldException) {
                e.printStackTrace()
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            }

//            val module = reactApplicationContext.getNativeModule(Timing::class.java)
//            try {
//                val isRunningTasks = Timing::class.java.getDeclaredField("isRunningTasks")
//                isRunningTasks.isAccessible = true
//                val v = isRunningTasks.get(module) as AtomicBoolean
//                v.set(true)
//            } catch (e: NoSuchFieldException) {
//                e.printStackTrace()
//            } catch (e: IllegalAccessException) {
//                e.printStackTrace()
//            }
        }
    }
}