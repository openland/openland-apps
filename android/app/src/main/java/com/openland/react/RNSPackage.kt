package com.openland.react

import android.view.View
import com.facebook.litho.config.ComponentsConfiguration
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ReactShadowNode
import com.facebook.react.uimanager.ViewManager
import com.openland.react.async.AsyncConfigManager
import com.openland.react.async.AsyncDataViewManager
import com.openland.react.async.AsyncListViewManager
import com.openland.react.async.AsyncViewManager
import com.openland.react.anim.RNSAnimatedViewManager
import com.openland.react.anim.RNSAnimatedViewViewManager
import com.openland.react.graphql.RNGraphQL
import com.openland.react.window.BottomSafeAreaProvider
import com.openland.react.window.RNSWindowManager

class RNSPackage : ReactPackage {

    init {
        ComponentsConfiguration.incrementalMountWhenNotVisible = false
    }

    override fun createNativeModules(reactContext: ReactApplicationContext): MutableList<NativeModule> {
        return mutableListOf(
                AsyncConfigManager(reactContext),
                AsyncDataViewManager(reactContext),
                RNSAnimatedViewManager(reactContext),
                RNSWindowManager(reactContext),
                BottomSafeAreaProvider(reactContext),
                BenchmarkModule(reactContext),
                RNGraphQL(reactContext)
        )
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): MutableList<ViewManager<View, ReactShadowNode<*>>> {
        return mutableListOf(
                AsyncViewManager(),
                AsyncListViewManager(),
                RNSAnimatedViewViewManager.sharedInstance
        ) as MutableList<ViewManager<View, ReactShadowNode<*>>>
    }
}