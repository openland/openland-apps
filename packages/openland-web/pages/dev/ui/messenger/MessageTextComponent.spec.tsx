import * as React from 'react';
import * as renderer from 'react-test-renderer';

jest.doMock('../components/DevDocsScaffold', () => {
    const ComponentToMock = ({ children }: any) => <div>{children}</div>;

    return {
        DevDocsScaffold: ComponentToMock,
    };
});

describe.only('MessageTextComponent', () => {
    const PageComponent = require('./MessageTextComponent.page').default;
    it('should render correctly', () => {
        expect(renderer.create(<PageComponent />).toJSON()).toMatchSnapshot();
    });
});
