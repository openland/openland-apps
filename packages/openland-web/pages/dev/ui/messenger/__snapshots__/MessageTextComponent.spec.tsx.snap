// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTextComponent should render correctly 1`] = `
@-webkit-keyframes animation_4vt6la {
  0%,
  80%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  40% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
@-moz-keyframes animation_4vt6la {
  0%,
  80%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  40% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
@-o-keyframes animation_4vt6la {
  0%,
  80%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  40% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
@keyframes animation_4vt6la {
  0%,
  80%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  40% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
@-webkit-keyframes animation_1qds50e {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-moz-keyframes animation_1qds50e {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-o-keyframes animation_1qds50e {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes animation_1qds50e {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-webkit-keyframes animation_1qxurr2 {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
@-moz-keyframes animation_1qxurr2 {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
@-o-keyframes animation_1qxurr2 {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
@keyframes animation_1qxurr2 {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
@-webkit-keyframes animation_yc61al {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
}
@-moz-keyframes animation_yc61al {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
}
@-o-keyframes animation_yc61al {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
}
@keyframes animation_yc61al {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
}
@-webkit-keyframes animation_17q4bf6 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-moz-keyframes animation_17q4bf6 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-o-keyframes animation_17q4bf6 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@keyframes animation_17q4bf6 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-webkit-keyframes animation_vxblz0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
}
@-moz-keyframes animation_vxblz0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
}
@-o-keyframes animation_vxblz0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
}
@keyframes animation_vxblz0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
}
@-webkit-keyframes animation_fv1i0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
}
@-moz-keyframes animation_fv1i0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
}
@-o-keyframes animation_fv1i0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
}
@keyframes animation_fv1i0 {
  0% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
}
@-webkit-keyframes animation_8loz2x {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
}
@-moz-keyframes animation_8loz2x {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
}
@-o-keyframes animation_8loz2x {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
}
@keyframes animation_8loz2x {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% 60%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% 60%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% 60%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% 60%;
  }
}
@-webkit-keyframes animation_w468rg {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-moz-keyframes animation_w468rg {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-o-keyframes animation_w468rg {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@keyframes animation_w468rg {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(1);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 50% calc(-10% + 11px);
    -webkit-transform: scale(0);
    -webkit-transform-origin: 50% calc(-10% + 11px);
  }
}
@-webkit-keyframes animation_17ugua6 {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
}
@-moz-keyframes animation_17ugua6 {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
}
@-o-keyframes animation_17ugua6 {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
}
@keyframes animation_17ugua6 {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 60% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 60% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 60% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 60% 50%;
  }
}
@-webkit-keyframes animation_1pv9uxs {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
}
@-moz-keyframes animation_1pv9uxs {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
}
@-o-keyframes animation_1pv9uxs {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
}
@keyframes animation_1pv9uxs {
  0% {
    opacity: 1;
    transform: scale(1);
    transform-origin: 40% 50%;
    -webkit-transform: scale(1);
    -webkit-transform-origin: 40% 50%;
  }
  100% {
    opacity: 0;
    transform: scale(0);
    transform-origin: 40% 50%;
    -webkit-transform: scale(0);
    -webkit-transform-origin: 40% 50%;
  }
}
@-webkit-keyframes animation_noxa33 {
  0% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
    display: none;
  }
  60% {
    display: block;
  }
  70% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
  }
  100% {
    opacity: 1;
    pointer-events: auto;
    visibility: auto;
  }
}
@-moz-keyframes animation_noxa33 {
  0% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
    display: none;
  }
  60% {
    display: block;
  }
  70% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
  }
  100% {
    opacity: 1;
    pointer-events: auto;
    visibility: auto;
  }
}
@-o-keyframes animation_noxa33 {
  0% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
    display: none;
  }
  60% {
    display: block;
  }
  70% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
  }
  100% {
    opacity: 1;
    pointer-events: auto;
    visibility: auto;
  }
}
@keyframes animation_noxa33 {
  0% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
    display: none;
  }
  60% {
    display: block;
  }
  70% {
    opacity: 0;
    pointer-events: none;
    visibility: hidden;
  }
  100% {
    opacity: 1;
    pointer-events: auto;
    visibility: auto;
  }
}
.css-h8iwlx,
[data-css-h8iwlx] {
  padding-left: 24px;
}
.css-10o9fv1,
[data-css-10o9fv1] {
  padding-right: 24px;
}
.css-1rxdtn0,
[data-css-1rxdtn0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: relative;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
}
.css-1rxdtn0 > *,
[data-css-1rxdtn0] > * {
  margin-top: 8px;
  margin-bottom: 8px;
}
.css-1rxdtn0 > :first-child,
[data-css-1rxdtn0] > :first-child {
  margin-top: 0px;
}
.css-1rxdtn0 > :last-child,
[data-css-1rxdtn0] > :last-child {
  margin-bottom: 0px;
}
.css-vlfxof,
[data-css-vlfxof] {
  height: 40px;
  color: #99a2b0;
  padding: 10px 16px;
  font-size: 14px;
  line-height: 20px;
  letter-spacing: 0;
  font-weight: 700;
}
.css-5uyuv6,
[data-css-5uyuv6] {
  flex-direction: column;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}
.css-1k8tu3j,
[data-css-1k8tu3j] {
  align-items: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}
.css-1owbun0,
[data-css-1owbun0] {
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}
.css-pf17gp,
[data-css-pf17gp] {
  color: #99a2b0;
}
.css-gqd39d,
[data-css-gqd39d] {
  cursor: pointer;
  background-color: #fff6e5;
  color: #1790ff;
}
.css-1f5rsnp,
[data-css-1f5rsnp] {
  cursor: pointer;
  color: #1790ff;
}

.css-0,
[data-css-0] {
  padding-left: 24px;
}

.css-1,
[data-css-1] {
  padding-right: 24px;
}

.css-2,
[data-css-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: relative;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
}

.css-2> *,
[data-css-2]> * {
  margin-top: 8px;
  margin-bottom: 8px;
}

.css-2>:first-child,
[data-css-2]>:first-child {
  margin-top: 0px;
}

.css-2>:last-child,
[data-css-2]>:last-child {
  margin-bottom: 0px;
}

.css-3,
[data-css-3] {
  height: 40px;
  color: #99a2b0;
  padding: 10px 16px;
  font-size: 14px;
  line-height: 20px;
  letter-spacing: 0;
  font-weight: 700;
}

.css-4,
[data-css-4] {
  flex-direction: column;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-5,
[data-css-5] {
  align-items: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-6,
[data-css-6] {
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

.css-7,
[data-css-7] {
  color: #99a2b0;
}

.css-8,
[data-css-8] {
  cursor: pointer;
  background-color: #fff6e5;
  color: #1790ff;
}

.css-9,
[data-css-9] {
  cursor: pointer;
  color: #1790ff;
}

<div>
  <div
    className="x css-0 css-1"
  >
    <div
      className="css-2"
    >
      <div
        className="css-3"
      >
        1. Joines
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             joined the room
          </span>
        </div>
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             joined the room along with 
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                Tara B
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             joined the room along with
             
            <span>
              <span
                className={
                  Array [
                    "
    cursor: pointer;
    color: #1790ff;
",
                  ]
                }
              >
                3
                 others
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        className="css-3"
      >
        2. Service Messages
      </div>
      <div
        className="css-3"
      >
        JobOpportunity
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             —
             
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             is interested in your job opportunity 
            <strong>
              “
              “Looking for React developer“
              “
            </strong>
             in
             
            <a
              className={
                Array [
                  "
    color: #1790ff;
",
                ]
              }
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              I am the walrus
            </a>
            .
            <br />
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             — as the next step, please, tell 
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             — a little bit about yourself.
          </span>
        </div>
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             —
             
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             is looking to recommend a candidate in response to your post 
            <strong>
              “
              “Looking for React developer“
              “
            </strong>
             in
             
            <a
              className={
                Array [
                  "
    color: #1790ff;
",
                ]
              }
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              I am the walrus
            </a>
            .
            <br />
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             — as the next step, please, describe your recommended candidate, how well do you know them, and share any relevant links.
          </span>
        </div>
      </div>
      <div
        className="css-3"
      >
        OfficeHours
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             —
             
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             is responding to your post
             
            <strong>
              “
              “Office hours with XX“
              “
            </strong>
             in 
            <a
              className={
                Array [
                  "
    color: #1790ff;
",
                ]
              }
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              I am the walrus
            </a>
            .
            <br />
            Now you can chat!
          </span>
        </div>
      </div>
      <div
        className="css-3"
      >
        RequestForStartups
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             —
             
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             is responding to your post
             
            <strong>
              “
              “Looking for React developer“
              “
            </strong>
             in 
            <a
              className={
                Array [
                  "
    color: #1790ff;
",
                ]
              }
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              I am the walrus
            </a>
            .
            <br />
            Now you can chat!
          </span>
        </div>
      </div>
      <div
        className="x css-4 css-5"
      >
        <div
          className="x css-6 css-5 css-7"
        >
          <span
            className={
              Array [
                "
    white-space: pre;
    text-align: center;
",
              ]
            }
          >
            🙌
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-8"
              >
                Sergey Lapin
              </span>
            </span>
             —
             
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <span
                className="css-9"
              >
                qwerty asdfgh
              </span>
            </span>
             is interested to make a recommendation following up to your post 
            <strong>
              “
              “Looking for React developer“
              “
            </strong>
             in
             
            <a
              className={
                Array [
                  "
    color: #1790ff;
",
                ]
              }
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              I am the walrus
            </a>
            .
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
